packages/trpc/server/routers/viewer/teams/publish.handler.ts

```diff
@@ -1,54 +1,139 @@
+import type { Prisma } from "@prisma/client";
+
 import { getRequestedSlugError } from "@calcom/app-store/stripepayment/lib/team-billing";
-import { purchaseTeamSubscription } from "@calcom/features/ee/teams/lib/payments";
+import {
+  purchaseTeamSubscription,
+  updateQuantitySubscriptionFromStripe,
+} from "@calcom/features/ee/teams/lib/payments";
 import { IS_TEAM_BILLING_ENABLED, WEBAPP_URL } from "@calcom/lib/constants";
+import { isOrganisationAdmin } from "@calcom/lib/server/queries/organisations";
 import { isTeamAdmin } from "@calcom/lib/server/queries/teams";
 import { closeComUpdateTeam } from "@calcom/lib/sync/SyncServiceManager";
 import { prisma } from "@calcom/prisma";
 import { teamMetadataSchema } from "@calcom/prisma/zod-utils";
 
 import { TRPCError } from "@trpc/server";
 
+import type { TrpcSessionUser } from "../../../trpc";
 import type { TPublishInputSchema } from "./publish.schema";
 
 type PublishOptions = {
   ctx: {
-    user: NonNullable<{ id: number }>;
+    user: NonNullable<TrpcSessionUser>;
```

Comment:
Heads up @alannnc @sean-brydon this line is breaking API builds: 
<img width="1089" alt="image" src="https://github.com/calcom/cal.com/assets/3504472/34b088d5-f764-4f74-a100-f5466893cacf">